domain/info.go

```diff
@@ -0,0 +1,220 @@
+func NewInfoSyncer(id string, etcdCli *clientv3.Client) *InfoSyncer {
+	return &InfoSyncer{
+		etcdCli:        etcdCli,
+		info:           getServerInfo(id),
+		serverInfoPath: fmt.Sprintf("%s/%s", ServerInformationPath, id),
+	}
+}
+
+// Init creates a new etcd session and stores server info to etcd.
+func (is *InfoSyncer) Init(ctx context.Context) error {
+	return errors.Trace(is.newSessionAndStoreServerInfo(ctx, owner.NewSessionDefaultRetryCnt))
+}
+
+// GetServerInfo gets self server static information.
+func (is *InfoSyncer) GetServerInfo() *ServerInfo {
+	return is.info
+}
+
+// GetServerInfoByID gets server static information from etcd.
+func (is *InfoSyncer) GetServerInfoByID(ctx context.Context, id string) (*ServerInfo, error) {
+	if is.etcdCli == nil || id == is.info.ID {
+		return is.info, nil
+	}
+	key := fmt.Sprintf("%s/%s", ServerInformationPath, id)
+	infoMap, err := getInfo(ctx, is.etcdCli, key, keyOpDefaultRetryCnt, keyOpDefaultTimeout)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	info, ok := infoMap[id]
+	if !ok {
+		return nil, errors.Errorf("[info-syncer] get %s failed", key)
+	}
+	return info, nil
+}
+
+// GetAllServerInfo gets all servers static information from etcd.
+func (is *InfoSyncer) GetAllServerInfo(ctx context.Context) (map[string]*ServerInfo, error) {
+	allInfo := make(map[string]*ServerInfo)
+	if is.etcdCli == nil {
+		allInfo[is.info.ID] = is.info
+		return allInfo, nil
+	}
+	allInfo, err := getInfo(ctx, is.etcdCli, ServerInformationPath, keyOpDefaultRetryCnt, keyOpDefaultTimeout, clientv3.WithPrefix())
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return allInfo, nil
+}
+
+// storeServerInfo stores self server static information to etcd.
+func (is *InfoSyncer) storeServerInfo(ctx context.Context) error {
+	if is.etcdCli == nil {
+		return nil
+	}
+	infoBuf, err := json.Marshal(is.info)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	err = ddl.PutKVToEtcd(ctx, is.etcdCli, keyOpDefaultRetryCnt, is.serverInfoPath, hack.String(infoBuf), clientv3.WithLease(is.session.Lease()))
+	return errors.Trace(err)
+}
+
+// RemoveServerInfo remove self server static information from etcd.
+func (is *InfoSyncer) RemoveServerInfo() {
+	if is.etcdCli == nil {
+		return
+	}
+	err := ddl.DeleteKeyFromEtcd(is.serverInfoPath, is.etcdCli, keyOpDefaultRetryCnt, keyOpDefaultTimeout)
+	if err != nil {
+		log.Errorf("[info-syncer] remove server info failed %v", err)
+	}
+}
+
+// Done returns a channel that closes when the info syncer is no longer being refreshed.
+func (is InfoSyncer) Done() <-chan struct{} {
+	if is.etcdCli == nil {
+		return make(chan struct{}, 1)
+	}
+	return is.session.Done()
+}
+
+// Restart restart the info syncer with new session leaseID and store server info to etcd again.
+func (is *InfoSyncer) Restart(ctx context.Context) error {
+	return errors.Trace(is.newSessionAndStoreServerInfo(ctx, owner.NewSessionRetryUnlimited))
```

Comment:
Why use NewSessionRetryUnlimited?

Reply:
I see in `schemaVersionSyncer. Restart()` use NewSessionRetryUnlimited when `Restart`, so...
How about your option?